File: Day1/part1.py
with open("./input.txt") as f:
    data = f.read().strip().split("\n")

sum = 0
for line in data:
    n = len(line)
    start_dig = 0
    end_dig = 0
    for i in range(n):
        if line[i] in "0123456789":
            start_dig = int(line[i])
            break

    for i in range(n - 1, -1, -1):
        if line[i] in "0123456789":
            end_dig = int(line[i])
            break

    sum += start_dig * 10 + end_dig

print(sum)

File: Day1/part1.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ifstream file("./Day1/input.txt");
    string line;
    int sum = 0;
    while (getline(file, line))
    {
        int start_dig = 0, end_dig = 0;
        for (int i = 0; i < line.size(); i++)
        {
            if (line[i] >= '0' && line[i] <= '9')
            {
                start_dig = line[i] - '0';
                break;
            }
        }
        for (int i = line.size() - 1; i >= 0; i--)
        {
            if (line[i] >= '0' && line[i] <= '9')
            {
                end_dig = line[i] - '0';
                break;
            }
        }
        sum += start_dig * 10 + end_dig;
    }
    cout << sum << endl;
}